### PR TITLE
Revert "[Scala 3] Revert type completions feature (#4236)"

### PR DESCRIPTION
**Pull request by tanishiking**
This reverts commit 6bf55ab632a900331720fd5f04e916596a3311f6 (PR https://github.com/scalameta/metals/pull/4236) that reverted the following PRs

- https://github.com/scalameta/metals/pull/4174
- https://github.com/scalameta/metals/pull/4223

### CODE REVIEW

**Comment by tanishiking** [comment]
FYI @dos65 @vzmerr

**Comment by vzmerr** [comment]
What do we do with this one? Do we merge it and then stash it with Vadim's PR when ready or wait for Vadim to rebase the PR on the current main and continue from then on? @dos65

**Comment by vzmerr** [comment]
It seems that Vadim's rework is not removing everything. It is changing them. So I guess this has to get merged anyway.

**Comment by dos65** [comment]
Yep. Let's merge it back

File: mtags/src/main/scala-2/scala/meta/internal/pc/AutoImportsProvider.scala [hunk 1]
```diff
@@ -21,9 +21,12 @@ final class AutoImportsProvider(
       filename = params.uri().toString(),
       cursor = Some(params.offset())
     )
+    typeCheck(unit)
```
Comment (dos65) [inline mtags/src/main/scala-2/scala/meta/internal/pc/AutoImportsProvider.scala:24]: looks like it should be removed

Comment (tanishiking) [reply mtags/src/main/scala-2/scala/meta/internal/pc/AutoImportsProvider.scala:24]: This is introduced by https://github.com/scalameta/metals/pull/4223

Comment (dos65) [reply mtags/src/main/scala-2/scala/meta/internal/pc/AutoImportsProvider.scala:24]: :thinking: cross test pass for scala2 without this line.
@tgodzik do you remember why it was needed?

Comment (tgodzik) [reply mtags/src/main/scala-2/scala/meta/internal/pc/AutoImportsProvider.scala:24]: och no, that might have been by mistake. Should be ok to remove it.

Comment (tanishiking) [reply mtags/src/main/scala-2/scala/meta/internal/pc/AutoImportsProvider.scala:24]: Ah, ok then let's remove this line :+1:

Comment (tanishiking) [reply mtags/src/main/scala-2/scala/meta/internal/pc/AutoImportsProvider.scala]: Removed

**Review by dos65 — APPROVED** [review]

**Pull request merged**